File: class23/java/AP2302-Exemplo02.java
import java.util.Scanner;
public class FuncaoPrimo {
    static boolean ehPrimo(int n) {
        if (n < 2) return false;
        for (int i = 2; i <= Math.sqrt(n); i++)
            if (n % i == 0) return false;
        return true;
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Digite um número: ");
        int n = sc.nextInt();
        if (ehPrimo(n)) System.out.println(n + " é PRIMO!");
        else System.out.println(n + " NÃO é primo.");
        sc.close();
    }
}

File: class23/python/AP2302-Exemplo02.py
# FuncaoPrimo — Aula 23
# Função que verifica se um número é primo

def eh_primo(n):
    if n < 2:
        return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True

num = int(input("Digite um número: "))
if eh_primo(num):
    print(f"{num} é PRIMO!")
else:
    print(f"{num} NÃO é primo.")
